sdk/python/src/buildkite_sdk/sdk.py

```diff
@@ -19,21 +21,55 @@ class Pipeline:
     def __init__(self):
         """A description of the constructor."""
         self.steps = []
+        self.agents = None
+        self.env = None
+        self.notify = None
         """I guess this is where we define the steps?"""
 
+    def add_agent(self, key: str, value: any):
+        if self.agents == None:
+            self.agents = {}
+        self.agents[key] = value
+
+    def add_environment_variable(self, key: str, value: any):
+        if self.env == None:
+            self.env = {}
+        self.env[key] = value
+
+    def add_notify(self, notify: PipelineNotify):
+        self.notify = notify
+
     def add_step(
         self,
         props: Union[
-            _block_step, _command_step, _group_step, _input_step, _trigger_step, _wait_step
+            _block_step,
+            _command_step,
+            _group_step,
+            _input_step,
+            _trigger_step,
+            _wait_step,
         ],
     ):
         """Add a command step to the pipeline."""
         self.steps.append(props.to_dict())
 
+    def to_pipeline(self):
```

Comment:
Can we name this similarly to what we name the other(s) (e.g., `minify` or `reduce` or whatever we decide to call it, as [suggested here](https://github.com/buildkite/buildkite-sdk/pull/28/files#r1994058875))?